dvc/repo/scm_context.py

```diff
@@ -3,8 +3,12 @@ def run(repo, *args, **kw):
         try:
             result = method(repo, *args, **kw)
             repo.scm.reset_ignores()
-            repo.scm.remind_to_track()
-            repo.scm.reset_tracked_files()
+            auto_stage = repo.config.get("core", {}).get("autostage", False)
+            if auto_stage:
+                repo.scm.track_changed_files()
+            else:
+                repo.scm.remind_to_track()
+                repo.scm.reset_tracked_files()
```

Comment:
A bit weird that `track_changed_files` calls it automatically, but here we do this explicitly. Better to stick to one approach or another for consistency.

Reply:
I agree, I'll just pull it out of `track_changed_files` for now. Maybe they would be better separate? I don't know :)